File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven, do: 40

  def remaining_minutes_in_oven(time_in_oven) do
    Lasagna.expected_minutes_in_oven - time_in_oven
  end

  def preparation_time_in_minutes(lasagna_layers), do: lasagna_layers*2

  def total_time_in_minutes(lasagna_layers, time_in_oven) do
    Lasagna.preparation_time_in_minutes(lasagna_layers) + time_in_oven
  end

  def alarm, do: "Ding!"
end

File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int elapsedTime) {
        return this.expectedMinutesInOven() - elapsedTime;
    }

    public int preparationTimeInMinutes(int layers) {
        return layers * 2;
    }

    public int totalTimeInMinutes(int layers, int elapsedTime) {
        return this.preparationTimeInMinutes(layers) + elapsedTime;
    }
}
